File: nse_scanner_ui/src/api/analytics.ts
import { api } from "./client"

export interface AnalysisTypeInfo {
  id: string
  label: string
  param_label: string
  min_default: number
  max_default: number
}

export interface AnalyticsConfig {
  types: AnalysisTypeInfo[]
}

export interface RunAnalysisRequest {
  analysis_type: string
  min_pct: number
  max_pct: number
}

export interface AnalyticsRow {
  symbol: string
  name: string | null
  sector: string | null
  ath: number
  ath_date: string
  last_close: number
  last_date: string
  pct_down_from_ath: number
}

export const analyticsApi = {
  config: () => api.get<AnalyticsConfig>("/api/analytics/config"),
  run: (body: RunAnalysisRequest) => api.post<AnalyticsRow[]>("/api/analytics/run", body),
}

File: nse_scanner/src/api/schemas/analytics.py
"""api/schemas/analytics.py — request/response models for the Analytics screen."""

from __future__ import annotations

from pydantic import BaseModel


class AnalysisTypeInfo(BaseModel):
    id: str
    label: str
    param_label: str
    min_default: float
    max_default: float


class AnalyticsConfig(BaseModel):
    types: list[AnalysisTypeInfo]


class RunAnalysisRequest(BaseModel):
    analysis_type: str
    min_pct: float
    max_pct: float
